recorder/src/main/cpp/perf_ctx.cc

```diff
@@ -9,6 +9,15 @@ PerfCtx::MergeSemantic PerfCtx::merge_semantic(PerfCtx::TracePt pt) {
     return static_cast<MergeSemantic>((pt >> PerfCtx::MERGE_SEMANTIC_SHIFT) & MERGE_SEMANTIC_MASK);
 }
 
+metrics::Mtr* s_m_bad_pairing;
+
+PerfCtx::ThreadTracker::ThreadTracker(Registry& _reg, ProbPct& _pct, int _tid) :
+    reg(_reg), pct(_pct), ignore_count(0), effective_start(0), effective_end(0), record(false), tid(_tid) {
+    effective.reserve((MAX_NESTING * (MAX_NESTING + 1)) / 2);
+}
+
+PerfCtx::ThreadTracker::~ThreadTracker() {}
+
 void PerfCtx::ThreadTracker::enter(PerfCtx::TracePt pt) {
```

Comment:
How about a counter to indicate ignored contexts when actual_stack.size() == PerfCtx::MAX_NESTING (conditional on line 34)